mitodl/open-discussions
# Added comment removal API

#### What are the relevant tickets?
Part of #278 

#### What's this PR do?
Updates the comments api to allow removal of comments by moderators.

Also reworked the `CommentFactory` because it hadn't been really used in a test yet and the previous iteration was impractical.

#### How should this be manually tested?
Go to `http://localhost:8063/api/v0/comments/:comment_id/` and `PATCH` with `{"removed": true}` and `{"removed": false}` and verify you see the expected state changes.
